corporate/views.py

```diff
@@ -49,9 +49,21 @@ def check_upgrade_parameters(
             raise BillingError('autopay with no card')
 
     min_licenses = seat_count
+    max_licenses = None
     if billing_modality == 'send_invoice':
         min_licenses = max(seat_count, MIN_INVOICED_LICENSES)
-    if licenses is None or licenses < min_licenses:
+        max_licenses = MAX_INVOICED_LICENSES
+
+    if licenses is None:
+        raise BillingError('not enough licenses',
+                           _("You must invoice for at least {} users.").format(min_licenses))
+
+    if max_licenses is not None and licenses > max_licenses:
+        message = _("Invoices with more than {} licenses can't be processed from this page. To complete the upgrade "
+                    "please contact {}.").format(max_licenses, settings.ZULIP_ADMINISTRATOR)
+        raise BillingError('too many licenses', message)
+
+    if licenses < min_licenses:
```

Comment:
I think this can still be an `or` gate with `licenses is None` to avoid duplication?